test/Dex.test.js

```diff
@@ -15,6 +17,169 @@ contract("IntegratedLimitOrderDex", (accounts) => {
   });
 
   it("should support placing sell orders", async () => {
-    await dex.sell(100, 10);
+    const oldBalance = (await dex.balanceOf.call(accounts[0]))
+    let tx = await dex.sell(100, 10);
+    const newBalance = (await dex.balanceOf.call(accounts[0]))
+
+    assert.equal(newBalance, oldBalance - 100, "Sell order did not update balance")
+    assert.equal(tx.logs[2].event, "NewSellOrder", "Incorrect event")
+    assert.equal(tx.logs[3].event, "OrderIncrease", "Incorrect event")
+  });
+  
+  it("should support placing multiple buy orders", async () => {
+    const amount = 1;
+    const oneEtherInWei = web3.utils.toWei("1");
+    const price = web3.utils.toBN(oneEtherInWei);
+    const numBuys = 5;
+    
+    let tx;
+    for (let i = 0; i < numBuys; i++) {
+      tx = await dex.buy(amount, price, {value: oneEtherInWei});
+    }
+
+    let orderQuantity = (await dex.getOrderQuantity(price))
+    assert.equal(orderQuantity, numBuys, "Buy order did not update balance")
+    assert.equal(tx.logs[0].event, "OrderIncrease", "Unexpected event emitted")
+  });
+  
+  it("should support fully filling sell order", async () => {
+    const oneEtherInWei = web3.utils.toWei("1");
+    const price = web3.utils.toBN(oneEtherInWei);
+    const amount = 5;
+    
+    let tx = await dex.sell(amount, price);
+    
+    let orderQuantity = (await dex.getOrderQuantity(price))
+    let balance = (await dex.getEthBalance(accounts[0]))
+    assert.equal(orderQuantity, 0, "Order quantity did not update balance")
+    assert.equal(balance, price * 5, "Ethereum balance did not update")
+  });
+  
+  it("should support partially filled sell orders", async () => {
+    const amount = 1;
+    const oneEthereInWei = web3.utils.toWei("1");
+    const price = web3.utils.toBN(oneEthereInWei);
+    const numBuys = 5;
+    let oldBalance = (await dex.getEthBalance(accounts[0]))
+
+    for (let i = 0; i < numBuys; i++) {
+      await dex.buy(amount, price, {value: oneEthereInWei});
+    }
+
+    const sellAmount = 4
+    await dex.sell(sellAmount, price);
+
+    let orderQuantity = (await dex.getOrderQuantity(price))
+    let balance = (await dex.getEthBalance(accounts[0]))
+    assert.equal(orderQuantity, numBuys - sellAmount, "Order quantity did not update balance")
+    assert.equal(balance - oldBalance, price * sellAmount, "Ethereum balance did not update")
+  });
+
+  it("should support placing multiple new sell orders", async () => {
+    const amount = 1;
+    const halfEtherInWei = web3.utils.toWei("0.5");
+    const price = web3.utils.toBN(halfEtherInWei);
+    let numSells = 3;
+    
+    let tx;
+    tx = await dex.sell(amount, price);
+    for (let i = 0; i < numSells; i++) {
+      await dex.sell(amount, price);
+    }
+
+    let orderQuantity = (await dex.getOrderQuantity(price))
+    assert.equal(orderQuantity, ++numSells, "Sell order did not update balance")
```

Comment:
Why is this ++numSells here? Do we need line 85 vs just 87? We can always check if i == 0, set tx. Really not a fan of inline incrementation, but I can accept it if you rather.

Reply:
Ah, line 85 is separated mainly because I wanted to extract the first transaction's events instead of the rest. I can do it your suggested way as well, will change in next push

Reply:
Thanks!